ansible/roles/configure/tasks/debian.yml

```diff
@@ -1,80 +1,86 @@
-  lineinfile:
+  ansible.builtin.lineinfile:
     path: /etc/ssh/sshd_config
     regexp: '^PubkeyAuthentication'
     line: 'PubkeyAuthentication yes'
 
-# Tasks for setting the hostname.
+# Task for setting the hostname.
 - name: "Setting the hostname."
-  hostname:
+  ansible.builtin.hostname:
     name: "localhost"
 
-# Tasks for restarting the SSH daemon.
+# Task for restarting the SSH daemon.
 - name: "Restarting the SSH daemon."
-  systemd:
+  ansible.builtin.systemd:
     name: ssh
     state: restarted
     daemon_reload: true
 
-# Tasks for disabling systemd-tmpfiles.
+# Task for disabling systemd-tmpfiles.
 - name: "Disabling systemd-tmpfiles."
-  replace:
+  ansible.builtin.replace:
     path: /usr/lib/tmpfiles.d/tmp.conf
     regexp: '^D'
     replace: '#D'
 
 # Tasks for configuring cloud-init.
 - name: "Configuring cloud-init."
+  when: enable_cloudinit == 'true'
   block:
     - name: "Message: Configuring cloud-init"
-      debug:
+      ansible.builtin.debug:
         msg: "Configuring cloud-init"
-    - file:
+    - name: "Remove unnecessary configuration files"
+      ansible.builtin.file:
         path: "{{ item }}"
         state: absent
       loop:
         - /etc/cloud/cloud.cfg.d/subiquity-disable-cloudinit-networking.cfg
         - /etc/cloud/cloud.cfg.d/99-installer.cfg
         - /etc/netplan/00-installer-config.yaml
       when: ansible_distribution == 'Ubuntu'
-    - blockinfile:
+    - name: "Ensure disable_vmware_customization is set to false in cloud.cfg"
+      ansible.builtin.blockinfile:
         path: /etc/cloud/cloud.cfg
         block: |
           disable_vmware_customization: false
-    - copy:
+    - name: "Set datasource_list in 90_dpkg.cfg"
+      ansible.builtin.copy:
         content: 'datasource_list: [ VMware, OVF, None ]'
         dest: /etc/cloud/cloud.cfg.d/90_dpkg.cfg
-  when: enable_cloudinit == 'true'
+        mode: '0644'
+
 
 - name: "Replacing dhcp with static for Debian"
-  lineinfile:
+  ansible.builtin.lineinfile:
     path: /etc/network/interfaces
     regexp: '^iface ens192 inet dhcp'
     state: absent
   when: enable_cloudinit == 'true' and ansible_distribution == 'Debian'
 
-# Tasks for setting SSH keys to regenerate.
+# Task for setting SSH keys to regenerate.
 - name: "Setting SSH keys to regenerate."
-  copy:
+  ansible.builtin.copy:
     dest: /etc/rc.local
     content: |
       #!/bin/bash
       if test -z "$(find /etc/ssh/ -iname 'ssh_host_*_key*')"; then
           dpkg-reconfigure openssh-server
       fi
       exit 0
-    mode: 0755
+    mode: '0755'
 
-# Tasks for force setting the guest operating system information for VMware Tools.
+# Task to force setting guest operating system information for VMware Tools.
 - name: "Setting guest operating system information for VMware Tools."
-  blockinfile:
+  ansible.builtin.blockinfile:
     path: /etc/vmware-tools/tools.conf
     block: |
       [guestosinfo]
       short-name = other5xLinux64Guest
```

Comment:
Im sure Im ignorant here, but what is the rational to not letting vmtools decide this, thereby showing us the distro flavour like Ubuntu or Debian etc?

Reply:
It's related to supporting some versions of Debian deployed using the content library on either v7 or v8 where it may lead to issues with guest customization. I can track down the old issue in the Issues.

Reply:
See https://github.com/vmware-samples/packer-examples-for-vsphere/pull/195#issuecomment-1219932995